### PR TITLE
Add explanation on how to run `cargo-clippy` and `clippy-driver`

### PR DESCRIPTION
**Pull request by Nilstrieb**
I don't know how this would be done on Windows or MacOS, feedback appreciated.

I did find this snippet for MacOS but it seems a little sketchy.

https://github.com/rust-lang/rust-clippy/blob/1d1e72308e5bcdf5abe95c1d29dfe82524940061/.github/workflows/clippy_bors.yml#L98-L102

changelog: none

### CODE REVIEW

File: book/src/development/adding_lints.md [hunk 1]
```diff
@@ -186,6 +186,13 @@ cargo dev lint input.rs
 from the working copy root. With tests in place, let's have a look at
 implementing our lint now.
 
+## Getting `cargo-clippy` and `clippy-driver` to work
```
Comment (giraffate) [inline book/src/development/adding_lints.md:189]: Can this be moved to https://github.com/rust-lang/rust-clippy/blob/master/book/src/usage.md#using-clippy-without-cargo-clippy-driver?

Comment (Nilstrieb) [reply book/src/development/adding_lints.md:189]: That looks like more of a user guide? This is only really for development

Comment (giraffate) [reply book/src/development/adding_lints.md:189]: Oh, you are right. I misunderstood it.

Question: In what situations would we use it directly during development?

Comment (Nilstrieb) [reply book/src/development/adding_lints.md:189]: I used it for perf testing so I could do `perf record cargo-clippy`. I'm not sure whether it's required but I have some problems with `cargo dev lint`. Although I had a lot of problems with everything to get everything to work with frame pointers :D. I personally also just like having direct access to stuff, but that's not that important.

**Review by giraffate — APPROVED** [review]
It looks good, thanks! Can the link be added to here: https://github.com/rust-lang/rust-clippy/pull/10531/files#diff-a5ade8b3f29bfda4754336da44290868956c0617b33e3dab780bee150c5d6208R21

**Comment by giraffate** [comment]
@bors r+

Thanks!

**Pull request merged**